subprojects/android-test/test-report/src/main/kotlin/com/avito/android/test/report/ReportImplementation.kt

```diff
@@ -355,33 +266,24 @@ class ReportImplementation(
 
     @Synchronized
     override fun addHtml(label: String, content: String) {
-        remoteStorage.upload(
+        val html = remoteStorage.upload(
             uploadRequest = RemoteStorage.Request.ContentRequest(
                 content = wrapInHtml(content),
                 extension = Entry.File.Type.html.name
             ),
             comment = label
-        ).let {
-            val currentState = state
-
-            if (currentState is ReportState.Initialized.Started && currentState.currentStep != null) {
-                currentState.currentStep!!.futureUploads.add(it)
-            } else {
-                earlyFuturesUploads.add(it)
-            }
-        }
+        )
+        val started = getCastedStateOrNull<ReportState.Initialized.Started>()
+        val futureUploads = started?.getCurrentStepOrCreate {
```

Comment:
And I get currentStep or create syntetic
But now I didn't call step callbacks. If I did for all entry would create a new synthetic step with a screenshot on each
If we need calling of those callbacks I will have to rewrite this logic